src/panels/lovelace/editor/card-editor/hui-card-picker.ts

```diff
@@ -8,60 +8,254 @@ import {
 } from "lit-element";
 import "@material/mwc-button";
 
+import "../../../../components/ha-card";
+
 import { HomeAssistant } from "../../../../types";
-import { LovelaceCardConfig } from "../../../../data/lovelace";
-import { CardPickTarget } from "../types";
+import { LovelaceCardConfig, LovelaceConfig } from "../../../../data/lovelace";
+import { CardPickTarget, CardPickerConfig } from "../types";
 import { fireEvent } from "../../../../common/dom/fire_event";
-import { getCardElementClass } from "../../create-element/create-card-element";
-
-const cards: string[] = [
-  "alarm-panel",
-  "conditional",
-  "entities",
-  "button",
-  "entity-filter",
-  "gauge",
-  "glance",
-  "history-graph",
-  "horizontal-stack",
-  "iframe",
-  "light",
-  "map",
-  "markdown",
-  "media-control",
-  "picture",
-  "picture-elements",
-  "picture-entity",
-  "picture-glance",
-  "plant-status",
-  "sensor",
-  "shopping-list",
-  "thermostat",
-  "vertical-stack",
-  "weather-forecast",
+import {
+  getCardElementClass,
+  createCardElement,
+} from "../../create-element/create-card-element";
+import { computeDomain } from "../../../../common/entity/compute_domain";
+
+import {
+  computeUnusedEntities,
+  computeUsedEntities,
+  EXCLUDED_DOMAINS,
+} from "../../common/compute-unused-entities";
+
+const cardConfigs: CardPickerConfig[] = [
```

Comment:
Should I move this to a new file or is there a better way to do this? Pulling information from each card configuration set on the card?

Reply:
We got to keep in mind that a custom card should also be able to add data to this, probably in a different array that is merged with this one so they can't screw with this one.

We do have a lot of redundant code now, we have `getStubConfig` that returns almost the same for config, but with the lazy loading that is harder to use in a card picker.

And we have <https://github.com/home-assistant/home-assistant-polymer/blob/dev/src/panels/lovelace/editor/card-editor/hui-dialog-edit-card.ts#L250-L260> that adds entities to a card config.

Since we want to give a suggestion to users when they start their card, I think we should combine all and create some helper functions. 

The card picker could also use the entities that are passed to the `edit-card-dialog` and advice cards based on these entities. <https://github.com/home-assistant/home-assistant-polymer/blob/dev/src/panels/lovelace/editor/card-editor/show-edit-card-dialog.ts#L8>

Reply:
Cool. Yea I'd like to make it more streamlined. I see that I can use those entities passed to the picker first and then used used and then used. Also, can make sure to send the data displayed on the picker into the card config.

Not sure what I would need to do to accommodate custom cards though.

Reply:
Do you think we should move the logic of getting the entity arrays (used, unused) to the edit card dialog? So that the card picker just have to choose from the array or filter it with the domains. And doesn't have to actually get the arrays

Reply:
Since we can already pass in an entity array to the dialog

Reply:
I am not sure how to use the `_params.entities` from the card picker. I was first thinking to send in the unused entities to the picker from the add card FAB. But once the card is selected and sent to the edit card dialog, it tries to update the config with all the unused entities. 

Need someway to update that param once the card is selected. Not sure the best way. Would it be to do it how it is? and then if entities exist update params.entities? 

or is it best to just add a qualifier to this  https://github.com/home-assistant/home-assistant-polymer/blob/dev/src/panels/lovelace/editor/card-editor/hui-dialog-edit-card.ts#L250-L260